### PR TITLE
Conditional require

### PR DESCRIPTION
**Pull request by waghanza**
As of `dpl` is **ruby** `1.9.3` compatible, dependencies management could be done either on by requiring a **ruby** `1.9.3` compatible version or requiring version depending on 
```ruby
RUBY_VERSION
```
is use

### CODE REVIEW

File: lib/dpl/provider/releases.rb [hunk 1]
```diff
@@ -3,7 +3,12 @@ class Provider
     class Releases < Provider
       require 'pathname'
 
-      requires 'octokit', version: '~> 4.3.0' # later versions require Ruby 2.x
+      if RUBY_VERSION >= "2.0.0"
+        requires 'octokit', version: '~> 4.6.2'
+      else
+        requires 'octokit', version: '~> 4.3.1'
```
Comment (BanzaiMan) [inline lib/dpl/provider/releases.rb]: Could you try `~> 4.3.0` here? https://travis-ci.org/travis-ci/dpl/jobs/185589375#L576

Comment (waghanza) [inline lib/dpl/provider/releases.rb]: done

**Comment by BanzaiMan** [comment]
Thanks for the updates. Could you test deployment with this provider and provide a link to the logs?

**Comment by waghanza** [comment]
sure, https://travis-ci.org/waghanza/dpl/builds/186407041 push this my `dpl` **fork** on https://github.com/waghanza/dpl/releases

**Comment by BanzaiMan** [comment]
@waghanza Please follow https://github.com/travis-ci/dpl/blob/master/TESTING.md to test this PR. Thanks.

**Comment by codesuki** [comment]
After rebasing ( https://github.com/codesuki/dpl/commits/conditional_require ) his branch on the current master I tried to test it here:
https://travis-ci.org/codesuki/dpl-github-test/builds/194682302

I set ruby to version 1.9.3 in the travis.yml but during dpl build it still uses a 2.x version.
Is there any way to change that?

**Comment by BanzaiMan** [comment]
Hmmm. The deployment appears to be happening with a newer version of Ruby, which is unexpected. We'll investigate.

**Comment by codesuki** [comment]
Maybe you can drop 1.9 support if that happens everywhere hahaha :)
Let me know what you find out and I will retest ;)

**Comment by BanzaiMan** [comment]
OK. So, we recently changed the way we select deployment ruby in anticipation of moving off 1.9.3 as the default Ruby on our build images. This behavior is expected (and I'd forgotten about it and was confused).

**Comment by codesuki** [comment]
OK! Thanks for the info. Glad this was merged.

**Pull request merged**